testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt

```diff
@@ -99,17 +101,34 @@ class SettingsMenuActivity : BaseActivity() {
         editAppId.addTextChangedListener(settingsTextWatcher)
         editSubscriptionKey.addTextChangedListener(settingsTextWatcher)
 
-        enableSaveView()
+        validateInputIDs()
     }
 
     private fun createSettingsInfo(): String {
         return "Build " + getString(R.string.miniapp_sdk_version) + " - " +
                 getString(R.string.build_version)
     }
 
-    private fun enableSaveView() {
-        saveViewEnabled =
-            !(editAppId.text.toString().isInvalidUuid() || editSubscriptionKey.text.isNullOrEmpty())
+    internal fun validateInputIDs() {
+        val isAppIdInvalid = editAppId.text.toString().isInvalidUuid()
+        val isSubsKeyInvalid = editSubscriptionKey.text.toString().includeHyphen().isInvalidUuid()
```

Comment:
I'm not sure we need to validate the subscription key as a UUID. The spec doesn't mention anything about it being a UUID, so I don't think we can rely on it always being that format.

The iOS SDK just validates that it's not empty ([see here](https://github.com/rakutentech/ios-miniapp/blob/master/Example/SettingsTableViewController.swift#L192)), so maybe that's enough?

Reply:
@corycaywood is right. The subscription key doesn't need to a UUID. It could be any string.